mypy/build.py

```diff
@@ -2696,12 +2715,15 @@ def load_graph(sources: List[BuildSource], manager: BuildManager,
         #   (since direct dependencies reflect the imports found in the source)
         #   but A's cached *indirect* dependency on C is wrong.
         dependencies = [dep for dep in st.dependencies if st.priorities.get(dep) != PRI_INDIRECT]
+        added = [dep for dep in st.suppressed
+                 if manager.find_module_cache.find_module(dep, manager.search_paths,
+                                                          manager.options.python_executable)]
         for dep in st.ancestors + dependencies + st.suppressed:
             # We don't want to recheck imports marked with '# type: ignore'
             # so we ignore any suppressed module not explicitly re-included
             # from the command line.
             ignored = dep in st.suppressed and dep not in entry_points
-            if ignored:
+            if ignored and dep not in added:
```

Comment:
My understanding is that the problem is that freshness of a file is determined by whether the suppressed deps are present _in the graph_. But they are never added to the graph in the first place because of this code (unless they are added as entry points). So I only add them (them == previously suppressed files that are now found) to the graph here in `load_graph`, and the invalidation is done by the code in `process_graph`.